### PR TITLE
Include additional cluster-wide trusted CA into Che

### PR DESCRIPTION
**Pull request by mmorhun**
Signed-off-by: Mykola Morhun <mmorhun@redhat.com>

### What does this PR do?
Includes CA certificates configured in `proxy/cluster` into list of trusted by Che certificates even if no proxy configuration added.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/17938

### How to test this PR?
1. Configure Openshift router with a custom certificate chain
2. Add all the CA certificates from the chain into `proxy/cluster` (excluding server one).
3. Deploy Che using operator from this branch
4. Start a workspace, ensure everything works as expected

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse/che-operator/blob/master/README.md#updating-custom-resource-definition-file)
- [ ] [Nightly OLM bundle is up to date](https://github.com/eclipse/che-operator/blob/master/README.md#update-nightly-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: pkg/controller/che/proxy.go [hunk 1]
```diff
@@ -48,14 +48,21 @@ func (r *ReconcileChe) getProxyConfiguration(checluster *orgv1.CheCluster) (*dep
...
 			return clusterWideProxyConf, nil
 		}
 
-		// proxy isn't configured
-		return &deploy.Proxy{}, nil
+		// Proxy isn't configured
+		// However add cluster-wide trusted CA certificates, if any, to Che
+		return &deploy.Proxy{
+			TrustedCAMapName: clusterWideProxyConf.TrustedCAMapName,
+		}, nil
```
Comment (tolusha) [inline pkg/controller/che/proxy.go:65]:
```suggestion
```

File: pkg/controller/che/proxy.go [hunk 2]
```diff
@@ -48,14 +48,21 @@ func (r *ReconcileChe) getProxyConfiguration(checluster *orgv1.CheCluster) (*dep
 			} else {
 				cheClusterProxyConf.NoProxy = deploy.MergeNonProxy(cheClusterProxyConf.NoProxy, ".svc")
 			}
+			// Add cluster-wide trusted CA certs
+			if clusterWideProxyConf.TrustedCAMapName != "" {
+				cheClusterProxyConf.TrustedCAMapName = clusterWideProxyConf.TrustedCAMapName
+			}
```
Comment (tolusha) [inline pkg/controller/che/proxy.go:54]:
```suggestion
			cheClusterProxyConf.TrustedCAMapName = clusterWideProxyConf.TrustedCAMapName
```

File: pkg/controller/che/proxy.go [hunk 3]
```diff
@@ -48,14 +48,21 @@ func (r *ReconcileChe) getProxyConfiguration(checluster *orgv1.CheCluster) (*dep
...
 			return cheClusterProxyConf, nil
 		} else if clusterWideProxyConf.HttpProxy != "" {
```
Comment (tolusha) [inline pkg/controller/che/proxy.go:56]:
```suggestion
		} else {
```

**Comment by tolusha** [comment]
add some tests to https://github.com/eclipse-che/che-operator/blob/master/pkg/controller/che/proxy_test.go

File: pkg/controller/che/proxy.go [hunk 4]
```diff
@@ -48,14 +48,19 @@ func (r *ReconcileChe) getProxyConfiguration(checluster *orgv1.CheCluster) (*dep
 			} else {
 				cheClusterProxyConf.NoProxy = deploy.MergeNonProxy(cheClusterProxyConf.NoProxy, ".svc")
 			}
+			// Add cluster-wide trusted CA certs, if any
+			cheClusterProxyConf.TrustedCAMapName = clusterWideProxyConf.TrustedCAMapName
 			return cheClusterProxyConf, nil
 		} else if clusterWideProxyConf.HttpProxy != "" {
 			clusterWideProxyConf.NoProxy = deploy.MergeNonProxy(clusterWideProxyConf.NoProxy, cheClusterProxyConf.NoProxy)
 			return clusterWideProxyConf, nil
 		}
```
Comment (tolusha) [inline pkg/controller/che/proxy.go:57]:
```suggestion
		} else {
			clusterWideProxyConf.NoProxy = deploy.MergeNonProxy(clusterWideProxyConf.NoProxy, cheClusterProxyConf.NoProxy)
			return clusterWideProxyConf, nil
		}
```

File: pkg/controller/che/proxy.go [hunk 5]
```diff
@@ -48,14 +48,19 @@ func (r *ReconcileChe) getProxyConfiguration(checluster *orgv1.CheCluster) (*dep
...
 
-		// proxy isn't configured
-		return &deploy.Proxy{}, nil
+		// Proxy isn't configured
+		// However add cluster-wide trusted CA certificates, if any, to Che
+		return &deploy.Proxy{
+			TrustedCAMapName: clusterWideProxyConf.TrustedCAMapName,
+		}, nil
```
Comment (tolusha) [inline pkg/controller/che/proxy.go:63]:
```suggestion
```

**Review by tolusha — APPROVED** [review]

**Review by AndrienkoAleksandr — APPROVED** [review]

**Comment by flacatus** [comment]
@mmorhun there are some job fixes in master

**Pull request merged**